front/src/applications/stdcm/components/StdcmForm/StdcmConsist.tsx

```diff
@@ -142,37 +154,52 @@ const StdcmConsist = ({ isDebugMode, disabled = false }: StdcmConsistProps) => {
   };
 
   useEffect(() => {
-    const totalMassError = validateTotalMass({
-      tractionEngineMass: rollingStock?.mass,
-      towedMass: towedRollingStock?.mass,
-      totalMass,
-    });
+    const totalMassError =
+      getMissingFieldMessage(totalMass) ??
+      validateTotalMass({
+        tractionEngineMass: rollingStock?.mass,
+        towedMass: towedRollingStock?.mass,
+        totalMass,
+      });
 
-    const totalLengthError = validateTotalLength({
-      tractionEngineLength: rollingStock?.length,
-      towedLength: towedRollingStock?.length,
-      totalLength,
-    });
+    const totalLengthError =
+      getMissingFieldMessage(totalLength) ??
+      validateTotalLength({
+        tractionEngineLength: rollingStock?.length,
+        towedLength: towedRollingStock?.length,
+        totalLength,
+      });
 
-    const maxSpeedError = validateMaxSpeed(maxSpeed, rollingStock?.max_speed);
+    const maxSpeedError =
+      getMissingFieldMessage(maxSpeed) ?? validateMaxSpeed(maxSpeed, rollingStock?.max_speed);
 
     // Hide the tooltip if the user just changed the value for an invalid one (should be displayed only on focus out)
     // but we also want to hide it as soon as the user corrects it
-    setConsistErrors((prevErrors) => ({
+    setConsistErrors({
```

Comment:
I think we can keep using `prevErrors` here and remove all consistErrors from the depths (you'll have to pass the true type of `setConsistErrors` in props to make it work)
```suggestion
    setConsistErrors((prevErrors) => ({...
```

Reply:
done